### DISCUSSION TITLE
Problème de merge. Besoins d'aide

### DISCUSSION THREAD

**Discussion by Baptisteismd** [community/community#86781] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Bonjour,
 Je débute avec GitHub et j’ai un projet. J’ai dû créer un fork sur le projet d’un autre groupe. J’ai créé une branche pour corriger les bugs, mais par erreur, j’ai fusionné une branche de mon fork dans le main de l’autre groupe et je ne sais pas comment revenir à la version précédente. Est-ce que quelqu’un saurait comment faire ? Merci beaucoup !

**Comment by faridnec** [comment]
Hey Bonjour

If you accidentally merged a branch from your fork into another group's main branch and you want to revert it you can do with some steps involving creating new branch. This approach creates a new commit that undoes the changes from the previous merge commit. It's a safer way to revert changes without rewriting history. But please make sure to back up your repository or create a new branch before proceeding just in case.

1. after you clone the repo (if you have done then skip). Use git log to find the commit hash that represents the merge commit you want to revert.
```bash
git log
```

2. Create a new branch to make changes without affecting the main branch. here the name is just an example. you can name it differently
```bash
git checkout -b revert-branch
```
3. Use git revert to create a new commit that undoes the changes introduced by the merge commit.
```bash
git revert -m 1 <commit_hash>
```
4. and push changes  to your fork on GitHub
```bash
git push origin revert-branch
```
5. Go to your fork on GitHub, and you should see an option to create a new pull request from the revert-branch to the other group's main branch.
6. Create the pull request, and once reviewed, merge it. This will revert the changes introduced by the merge commit.
